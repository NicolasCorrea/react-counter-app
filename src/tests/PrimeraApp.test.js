import React from "react";
import { shallow } from "enzyme";
import  '@testing-library/jest-dom';
import PrimeraApp from "../PrimeraApp";

describe('test on PrimeraApp', () => {
    // test('should show message "hola soy goku"', () => {

    //     const saludo = 'hola soy goku';

    //     const wrapper = render(<PrimeraApp saludo={saludo} />);

    //     expect(wrapper.getByText(saludo)).toBeInTheDocument();
        

    // })

    test('should show PrimeraApp correct', () => {
        const saludo = 'hola soy goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);
        
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show the subtitle send by props', () => {
        const saludo = 'hola soy goku';
        const subtitle = 'soy un subtitle';
        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitle}/>);
        
        const textoParrafo = wrapper.find("p").text();
    
        expect(textoParrafo).toBe(subtitle);
        
    })
    
    
})
