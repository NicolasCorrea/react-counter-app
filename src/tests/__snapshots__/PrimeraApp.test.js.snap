// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on PrimeraApp should show PrimeraApp correct 1`] = `
<Fragment>
  <h1>
    hola soy goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
